### PR TITLE
Add config to set max request body size to save HTTP POST parameters

### LINKED ISSUE #28040: Fix possible DoS issue (CVE-2024-22353)
IBM WebSphere Application Server Liberty is vulnerable to a denial of service (CVE-2024-22353 CVSS 5.9).

### PR DESCRIPTION
**Pull request by ayoho**
- Adds a `postParamMaxRequestBodySize` attribute to the `<webAppSecurity>` element
- Limits POST request parameters being saved to requests with Content-Lengths < 128 MB, by default.

Fixes #28040

### CODE REVIEW

**Comment by ayoho** [comment]
#build

**Comment by clarkek123** [comment]
Fixes #PH59146

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -50,6 +47,9 @@ postParamSaveMethod.desc=Specifies where POST parameters are stored upon redirec
 postParamCookieSize=POST parameter cookie size
 postParamCookieSize.desc=Size of the POST parameter cookie. If the size of the cookie is larger than the browser limit, unexpected behavior may occur. The value of this property must be a positive integer and represents the maximum size of the cookie in bytes. The equivalent custom property in the full application server profile is com.ibm.websphere.security.util.postParamMaxCookieSize.
 
+postParamMaxRequestBodySize=Maximum request body size to save POST parameters
+postParamMaxRequestBodySize.desc=Specifies the maximum size of the request body of an HTTP POST request to save request parameters. If the content length of the request is larger than the maximum allowed size, any POST parameters in the request will not be saved. If the content length of the request is unknown, request parameters will not be saved regardless of the value of this attribute. Setting this attribute to -1 means that POST parameters will always be saved as long as the content length of the request is known.
```
Comment (helyarp) [inline dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]: - Change "will not be" to "are not"
- Replace "will ... be" and "as long as" (presumably, the "long" doesn't refer to length) in the last sentence, for example: "Setting this attribute to -1 means that POST parameters are always saved if the content length of the request is known."

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -50,6 +47,9 @@ postParamSaveMethod.desc=Specifies where POST parameters are stored upon redirec
...
+
 allowLogoutPageRedirectToAnyHost=Allow logout page to redirect to any host
 allowLogoutPageRedirectToAnyHost.desc=Warning, security risk: Setting this property to true may open your systems to potential URL redirect attacks. If set to true, any host can be specified for the logout page redirect. If set to false, and the logout page points to a different host, or one not listed in the logout page redirect domain list, then a generic logout page is displayed. The equivalent custom property in the full application server profile is com.ibm.websphere.security.allowAnyLogoutExitPageHost.
```
Comment (helyarp) [inline dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties:54]: Change "may" to "might"

**Review by helyarp — COMMENTED** [review]
Suggested changes

**Comment by ayoho** [comment]
#libby
#build

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (helyarp) [reply dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]: Pls change "will not be" to "are not" in the desc. (There were 2 "will not be" instances. One was changed.)

Comment (ayoho) [reply dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]: Missed that one, sorry.

**Comment by ayoho** [comment]
#build

**Review by helyarp — COMMENTED** [review]
Looks good!

**Comment by ayoho** [comment]
#build

**Comment by ayoho** [comment]
Green build with both OL and CL changes together: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_yQyAsPK7Ee6Y5NNrWk9AvQ

**Comment by ayoho** [comment]
#build

**Comment by ayoho** [comment]
#build
#libby

**Review by arunavemulapalli — APPROVED** [review]

**Pull request merged**